Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Collections/GridObjectCollection.cs

```diff
@@ -455,15 +487,23 @@ public void OnValidate()
                     // Try to guess what the desired columns would be
                     int columnsGuess = Mathf.CeilToInt((float)nodeListCount / rows);
                     string objectName = gameObject.name;
-                     if (gameObject.transform.parent != null) 
-                     {
-                         objectName += " (parent " + transform.parent.gameObject.name + ")";
-                     }
-                    Debug.Log("GridObjectCollection on " + objectName + " has layout ColumnsThenRows but columns are not specified. Most likely from asset upgrade to MRTK 2.2. Settings Columns property to " + nodeListCount + "/ " + rows + " = " + columnsGuess +". Check your asset to make sure GridObjectCollection has the correct values.");
-                    Columns = columnsGuess;
+                    if (gameObject.transform.parent != null)
+                    {
+                        objectName += " (parent " + transform.parent.gameObject.name + ")";
+                    }
+                    Debug.Log("Asset update to MRTK 2.2 on GridObjectCollection on " + objectName 
+                        + " layout is ColumnsThenRows, columns is " + columns
+                        + " rows is " + rows +
+                        ". Updating columns to " + nodeListCount + "/ " + rows + " = " + columnsGuess 
+                        + " and layout out grid. Check asset to make sure GridObjectCollection has the correct values.");
+                    Undo.RecordObject(this, "update rows columns");
```

Comment:
Looks like you have undo.recordobject twice?

Reply:
Good catch!